TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -15,7 +15,52 @@ public CodeElement CreateCommitStatement(CodeElementsParser.CommitStatementConte
         {
             return new CommitStatement();
         }
+        public CodeElement CreateTruncateStatement(CodeElementsParser.TruncateStatementContext context)
+        {
+            var tableName = CreateTableOrViewOrCorrelationName(context.tableName);
+            var storageManagementClause =
+                CreateStorageManagementClause(context.storageManagementClause());
+            var deleteTriggersHandlingClause = CreateDeleteTriggersHandlingClause(context.deleteTriggersHandlingClause());
+            var isImmediate = new SyntaxProperty<bool>(context.SQL_IMMEDIATE()!=null,null);
+            return new TruncateStatement(tableName, storageManagementClause, deleteTriggersHandlingClause,isImmediate);
+        }
+
+        public SyntaxProperty<TruncateStatement.StorageManagementOption> CreateStorageManagementClause(CodeElementsParser.StorageManagementClauseContext context)
+        {
+            SyntaxProperty<TruncateStatement.StorageManagementOption> storageManagement = null;
+            if (context.reuse() != null)
+            {
+                storageManagement = new SyntaxProperty<TruncateStatement.StorageManagementOption>(TruncateStatement.StorageManagementOption.ReuseStorage,context.reuse().KeywordREUSE as Token);
```

Comment:
You can use ParseTreeUtils.GetFirstToken(context) to get the reuse token easily. Check how it's done for Cobol in similar cases.

`.KeywordREUSE as Token` is somewhat ambiguous because we know that this is a `Token` instance. Besides it is better to keep the same code between builders, it helps the reader.

Please change it in the whole file.